R/object2fixed.R

```diff
@@ -0,0 +1,93 @@
+
+#' Match quanteda objects against token types
+#'
+#' Developer function to match patterns in quanteda objects against token types.
+#' @param x a list of character vectors, [dictionary] or collocations object
+#' @inheritParams valuetype
+#' @inheritParams pattern2id
+#' @param concatenator the concatenation character that join multi-word
+#'   expression in `types`
+#' @inheritParams tokens_lookup
+#' @param remove_unigram  if `TRUE`, ignores single-word patterns
+#' @return a list of integer vectors containing indices of matched types
+#' @seealso [pattern2id()]
+#' @keywords development
+#' @export
+object2id <- function(x, types, valuetype = c("glob", "fixed", "regex"),
+                      case_insensitive = TRUE,
+                      concatenator = "_", levels = 1, remove_unigram = FALSE,
+                      keep_nomatch = FALSE) {
+    
+    if (is.dfm(x))
+        stop("dfm cannot be used as pattern")
+    
+    types <- check_character(types, min_len = 0, max_len = Inf, strict = TRUE)
+    valuetype <- match.arg(valuetype)
+    case_insensitive <- check_logical(case_insensitive)
+    concatenator <- check_character(concatenator)
+    levels <- check_integer(levels, min = 1, max_len = Inf)
+    remove_unigram <- check_logical(remove_unigram)
+    keep_nomatch <- check_logical(keep_nomatch)
+    
+    if (is.collocations(x)) {
```

Comment:
Do you think this will be easier to maintain than using method dispatch for specific object classes of `x`?